test/node-fetch/main.js

```diff
@@ -529,143 +511,144 @@ describe('node-fetch', () => {
         return chunks
       }
 
-      return expect(read(res.body))
-        .to.eventually.be.rejectedWith(Error)
+      return assert.rejects(read(res.body), new TypeError('terminated'))
     })
   })
 
   it('should handle network-error in chunked response in consumeBody', () => {
     const url = `${base}error/premature/chunked`
     return fetch(url).then(res => {
-      expect(res.status).to.equal(200)
-      expect(res.ok).to.be.true
+      assert.strictEqual(res.status, 200)
+      assert.strictEqual(res.ok, true)
 
-      return expect(res.text()).to.eventually.be.rejectedWith(Error)
+      return assert.rejects(res.text(), new TypeError('terminated'))
     })
   })
 
   it('should handle DNS-error response', () => {
     const url = 'http://domain.invalid'
-    return expect(fetch(url)).to.eventually.be.rejectedWith(TypeError)
+    return assert.rejects(fetch(url), new TypeError('fetch failed'))
   })
 
+  // TODO: Should we pass through the error message?
   it('should reject invalid json response', () => {
     const url = `${base}error/json`
     return fetch(url).then(res => {
-      expect(res.headers.get('content-type')).to.equal('application/json')
-      return expect(res.json()).to.eventually.be.rejectedWith(Error)
+      assert.strictEqual(res.headers.get('content-type'), 'application/json')
+      return assert.rejects(res.json(), SyntaxError)
```

Comment:
this exposes some information about the invalid json. Should we do that (Chrome does it too)